scala-package/infer/src/main/scala/org/apache/mxnet/infer/ImageClassifier.scala

```diff
@@ -182,7 +180,7 @@ object ImageClassifier {
     resizedImage.flush()
 
     // creating NDArray according to the input shape
-    val pixelsArray = NDArray.array(result, shape = inputImageShape)
+    val pixelsArray = NDArray.array(result, shape = new Shape(1 +: inputImageShape.toVector))
```

Comment:
@nswamy as we discussed offline, this is just a design decision. As an alternative, we will request the user to add a reshape if they would like to concatenate the Image NDArray altogether in one piece.

Reply:
The forward operation most often contains a axis for Batch, either you can do it here or outside the method just before you call forward. I think this is fine or if you guys think we should outside that is also fine -- you can use the NDArray.expand_dims() if it is a single dimension
The earlier version was wrong it was expecting the batch axis to be already there.

Reply:
I think it should be done outside. Currently it does exactly what the javadocs say it does. Takes a buffered image and turns it into an NDArray of shape inputShape. As it turns out this isn't actually what we wanted to be doing.

Also, this is a public api. As long as we have a reasonable alternative we should try to avoid changing it's behavior.